csrc/preseg_passes/insert_reshardings.cpp

```diff
@@ -117,6 +117,28 @@ void insertReshardingsAfter(Fusion* fusion) {
 } // namespace
 
 void InsertReshardingsPass::runPass(Fusion* fusion) {
+  for (TensorView* tv : fusion->allTvs()) {
+    std::vector<int64_t> rfactor_axes;
+    rfactor_axes.reserve(tv->nDims());
+
+    for (auto&& [i, loop_id] : enumerate(tv->getLoopDomain())) {
+      if (loop_id->isReduction()) {
+        if (std::count(
+                tv->getLogicalDomain().begin(),
+                tv->getLogicalDomain().end(),
+                loop_id) == 0) {
```

Comment:
Is this just to separate logical DID split from loop DID split?

Reply:
I hope my new code comments help.